arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -1699,19 +1712,24 @@ Result RestReplicationHandler::processRestoreDataBatch(transaction::Methods& trx
       if (type == REPLICATION_MARKER_DOCUMENT) {
         VPackSlice const doc = marker.get(::dataString);
         TRI_ASSERT(doc.isObject());
-        if (isUsersOnCoordinator) {
-          // In the _users case we silently remove the _key value.
-          builder.openObject();
-          for (auto it : VPackObjectIterator(doc)) {
-            if (arangodb::velocypack::StringRef(it.key) != StaticStrings::KeyString &&
-                arangodb::velocypack::StringRef(it.key) != StaticStrings::IdString) {
-              builder.add(it.key);
-              builder.add(it.value);
-            }
+        // In the _users case we silently remove the _key value.
+        VPackObjectBuilder guard(&builder);
+        for (auto it : VPackObjectIterator(doc)) {
+          if (isUsersOnCoordinator &&
+              (arangodb::velocypack::StringRef(it.key) == StaticStrings::KeyString &&
```

Comment:
I think this is not the negation of the original case. Now there is a continue instead of adding the doc. Please check if the logic really does what you expect. Originally the rule hit if neither key or id was in the string ref. This condition is likely to be never true. If this was meant to be the negation, then should `builder.add(doc)` be used here?

Reply:
Good catch, it should have been `||` instead of `&&`.